.github/workflows/self-nightly-past-ci-caller.yml

```diff
@@ -0,0 +1,140 @@
+name: Self-hosted runner (nightly-past-ci-caller)
+

```

Comment:
New workflow file to run nightly CI.
(This is extracted from the existing workflow file `self-scheduled.yml`)

Mainly to be triggered from `self-nightly-past-ci-caller.yml` via `workflow_cal`l event.

The name is a bit misleading - the nightly CI and the past CI are chained in `self-nightly-past-ci-caller.yml`, and they won't be run in a daily basis (as one round is too long to run in a single day).